fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/services/VersionServiceImpl.java

```diff
@@ -206,28 +228,67 @@ private Triple convertToInternalReference(final Triple t,
     }
 
     @Override
-    public FedoraResource createBinaryVersion(final FedoraSession session,
-            final FedoraResource resource,
+    public FedoraBinary createBinaryVersion(final FedoraSession session,
+            final FedoraBinary resource,
+            final Instant dateTime,
+            final StoragePolicyDecisionPoint storagePolicyDecisionPoint)
+            throws InvalidChecksumException {
+        return createBinaryVersion(session, resource, dateTime, null, null, storagePolicyDecisionPoint);
+    }
+
+    @Override
+    public FedoraBinary createBinaryVersion(final FedoraSession session,
+            final FedoraBinary resource,
             final Instant dateTime,
             final InputStream contentStream,
-            final String filename,
-            final String mimetype,
-            final Collection<URI> checksums) {
+            final Collection<URI> checksums,
+            final StoragePolicyDecisionPoint storagePolicyDecisionPoint) throws InvalidChecksumException {
```

Comment:
I am not seeing where `storagePolicyDecisionPoint` is used. Is there any reason not to remove it from the interface?

Reply:
I don't actually know what its used for, but that was unintentional, it's being passed along to setContent now.